tests/backup/backup_kubevirt_test.go

```diff
@@ -413,3 +413,209 @@ var _ = Describe("{KubevirtVMBackupRestoreWithDifferentStates}", func() {
 		CleanupCloudSettingsAndClusters(backupLocationMap, cloudCredName, cloudCredUID, ctx)
 	})
 })
+
+// This testcase verifies backup and restore of Kubevirt VMs after upgrading kubevirt version
+var _ = Describe("{KubevirtUpgradeTest}", func() {
+
+	var (
+		restoreNames         []string
+		backupPreUpgrade     string
+		backupPostUpgrade    string
+		scheduledAppContexts []*scheduler.Context
+		sourceClusterUid     string
+		cloudCredName        string
+		cloudCredUID         string
+		backupLocationUID    string
+		backupLocationName   string
+		backupLocationMap    map[string]string
+		providers            []string
+		labelSelectors       map[string]string
+		namespaceMapping     map[string]string
+	)
+
+	JustBeforeEach(func() {
+		StartPxBackupTorpedoTest("KubevirtUpgradeTest", "Verify backup and restore of Kubevirt VMs after upgrading Kubevirt control plane", nil, 93013, Mkoppal, Q3FY24)
+
+		backupLocationMap = make(map[string]string)
+		labelSelectors = make(map[string]string)
+		providers = getProviders()
+		namespaceMapping = make(map[string]string)
+
+		log.InfoD("scheduling applications")
+		scheduledAppContexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			taskName := fmt.Sprintf("%d-%d", 93013, i)
+			appContexts := ScheduleApplications(taskName)
+			for _, appCtx := range appContexts {
+				appCtx.ReadinessTimeout = appReadinessTimeout
+				scheduledAppContexts = append(scheduledAppContexts, appCtx)
+			}
+		}
+	})
+
+	It("Verify backup and restore of Kubevirt VMs in different states", func() {
+		defer func() {
+			log.InfoD("switching to default context")
+			err := SetClusterContext("")
+			log.FailOnError(err, "failed to SetClusterContext to default cluster")
+		}()
```

Comment:
JustAfterEach will handle this, right?

Reply:
In `JustAfterEach` also it is deferred. So it will run after `JustAfterEach`. We will need it here as well.